sale_order_min_qty/models/sale.py

```diff
@@ -0,0 +1,56 @@
+# Copyright 2019 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.constrains("order_line")
```

Comment:
IMHO you need to put constraint on order line level as this will trigger on order_line creation. But if user modifies qty on existing one, the constraint will not raise.